src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -148,15 +150,17 @@ applyCmd
     -> Bool
       -- ^ execution config for module install
     -> IO (T2 (CommandResult [TxLog Value]) ModuleCache)
-applyCmd logger pdbenv miner gasModel pd spv cmdIn mcache0 ecMod =
+applyCmd v logger pdbenv miner gasModel pd spv cmdIn mcache0 ecMod =
     second _txCache <$!>
       runTransactionM cenv txst applyBuyGas
   where
     txst = TransactionState mcache0 mempty 0 Nothing (_geGasModel freeGasEnv)
 
     executionConfigNoHistory = mkExecutionConfig $
       [ FlagDisableHistoryInTransactionalMode ] ++
-      if ecMod then [] else [ FlagDisableModuleInstall ]
+      [ FlagDisableModuleInstall | not ecMod ] ++
+      [ FlagOldReadOnlyBehavior | isPactBackCompatV16 ]
```

Comment:
What does `FlagOldReadOnlyBehavior` do?

Reply:
It's what fixes `enforce-one`, which attempts "full read only" within `enforce-guard`'s  "system read only". Before this resulted in an error, now `runReadOnly` runs with reduced privileges and does not fail (unless of course what's being run hits a user table)

Reply:
See https://github.com/kadena-io/pact/blob/master/src/Pact/Types/Purity.hs#L135

Reply:
Should we add a line in https://pact-language.readthedocs.io/en/latest/pact-reference.html#pure-execution about the different types of purity present in pact (i.e. "full read only" vs. "system read only")?